synapse/types.py

```diff
@@ -83,33 +83,32 @@ class ISynapseReactor(
     """The interfaces necessary for Synapse to function."""
 
 
-class Requester(
-    namedtuple(
-        "Requester",
-        [
-            "user",
-            "access_token_id",
-            "is_guest",
-            "shadow_banned",
-            "device_id",
-            "app_service",
-            "authenticated_entity",
-        ],
-    )
-):
+@attr.s
```

Comment:
Should this use slots / be frozen? (I suspect it can be frozen OK since tuples aren't able to be modified?)

Reply:
yes it should. done.